### PR TITLE
refactor Registry Form

### PR DESCRIPTION
**Pull request by David-DB88**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Wizard registry form refactoring
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaCluster/KafkaCluster.tsx [hunk 1]
```diff
@@ -54,7 +54,7 @@ const KafkaCluster: React.FC = () => {
         <InputHint>
           the list of Kafka brokers that you want to connect to
         </InputHint>
-        <S.BootstrapServersContainer>
+        <S.Container>
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaCluster/KafkaCluster.tsx:57]: Container of what? I would suggest something like `ArrayFieldWrapper`

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx [hunk 2]
```diff
@@ -5,104 +5,117 @@ import Input from 'components/common/Input/Input';
...
+  } = useFormContext();
+
+  const showRegistryFrom: React.MouseEventHandler<HTMLButtonElement> = (e) => {
     e.preventDefault();
-    setNewSchemaRegistry(!newSchemaRegistry);
-    if (!newSchemaRegistry) {
-      methods.reset({ url: '' });
+    setIsShowRegistryForm(!isShowRegistryForm);
+    if (!isShowRegistryForm) {
+      reset({
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:25]: setValue. We should not reset form

Comment (workshur) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:25]: Reset leads to updating of default value it is a wrong way

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx [hunk 3]
```diff
@@ -5,104 +5,117 @@ import Input from 'components/common/Input/Input';
...
 import { ErrorMessage } from '@hookform/error-message';
 import { useFormContext } from 'react-hook-form';
+import Heading from 'components/common/heading/Heading.styled';
+import { InputLabel } from 'components/common/Input/InputLabel.styled';
 
 const SchemaRegistry = () => {
-  const [newSchemaRegistry, setNewSchemaRegistry] = useState(false);
-  const methods = useFormContext();
-  const registry: React.MouseEventHandler<HTMLButtonElement> = (e) => {
+  const [isShowRegistryForm, setIsShowRegistryForm] = useState(true);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:12]: Just watch `schemaRegistry` field. If it is empty(undefined) show nothing

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx [hunk 4]
```diff
@@ -5,104 +5,117 @@ import Input from 'components/common/Input/Input';
...
-                  errors={methods.formState.errors}
-                  name="schemaRegistry.url"
-                />
+                <ErrorMessage errors={errors} name="schemaRegistry.url" />
               </FormError>
-            </S.PartStyled>
-            <S.PartStyled>
+            </div>
+            <div>
               <S.CheckboxWrapper>
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:65]: Check how I use checkboxes

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx [hunk 5]
```diff
@@ -5,104 +5,117 @@ import Input from 'components/common/Input/Input';
...
           buttonType="primary"
-          onClick={(e) => registry(e)}
+          onClick={(e) => showRegistryFrom(e)}
         >
-          {!newSchemaRegistry ? 'Add Schema Registry' : 'Remove from config'}
+          {!isShowRegistryForm ? 'Add Schema Registry' : 'Remove from config'}
         </Button>
-        {newSchemaRegistry && (
+      </div>
+      <S.Container>
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:49]:
```suggestion
      <S.ArrayFieldContainer>
```

File: kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.tsx [hunk 6]
```diff
@@ -14,10 +14,17 @@ type BootstrapServer = {
...
+  url: string;
+  isAuth: boolean;
+  username: string;
+  password: string;
+};
 export type FormValues = {
   name: string;
   readOnly: boolean;
   bootstrapServers: BootstrapServer[];
+  schemaRegistry: SchemaRegistryType;
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.tsx:27]: It should be optional field I believe

**Review by workshur — CHANGES_REQUESTED** [review]

File: kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaCluster/KafkaCluster.tsx (hunk 1 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/KafkaCluster/KafkaCluster.tsx:57]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx (hunk 2 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:25]: we can add just {keepDefaultValue: true},
but i will do it with setValue

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx (hunk 3 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:12]: I didn't understand very well what you mean,
something like this ?

  const [isShowRegistryForm, setIsShowRegistryForm] = useState(!watch('schemaRegistry'));

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx (hunk 5 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:49]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.tsx (hunk 6 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/WizardForm.tsx:27]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx (hunk 4 above)
Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:65]: ok

File: kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx (hunk 3 above)
Comment (Mgrdich) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:12]: i think what he meant is that `const isShowRegistryForm = watch('schemaRegistry')`, cause as far as it is directly related to the `schemaRegistry` field , so you don't need extra setter on the that level.

Comment (David-DB88) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx:12]: we need not only reset the form, also we need to close form with clicking button, so just watch schemaRegistry i think not enough.

Comment (Mgrdich) [reply kafka-ui-react-app/src/components/Wizard/WizardForm/SchemaRegistry/SchemaRegistry.tsx]: yeah that might not work during non empty default values.

**Pull request merged**